libraries/ms-helper/src/sequenced_relay.c

```diff
@@ -1,33 +1,25 @@
-#include "motor_relay.h"
+#include "sequenced_relay.h"
 #include <string.h>
 #include "can_msg_defs.h"
 #include "exported_enums.h"
 #include "soft_timer.h"
 
 static void prv_delay_cb(SoftTimerID timer_id, void *context) {
-  MotorRelayStorage *storage = context;
+  SequencedRelayStorage *storage = context;
 
+  storage->delay_timer = SOFT_TIMER_INVALID_TIMER;
   // We only bother with the delay if we're closing the relays, so assume it's closing
   gpio_set_state(&storage->settings.right_relay, GPIO_STATE_HIGH);
 }
 
 static StatusCode prv_handle_relay_rx(SystemCanMessage msg_id, uint8_t state, void *context) {
-  MotorRelayStorage *storage = context;
-
-  soft_timer_cancel(storage->delay_timer);
+  SequencedRelayStorage *storage = context;
 
-  gpio_set_state(&storage->settings.left_relay, state);
-  if (state == EE_RELAY_STATE_CLOSE) {
-    soft_timer_start_millis(storage->settings.delay_ms, prv_delay_cb, storage,
-                            &storage->delay_timer);
-  } else {
-    gpio_set_state(&storage->settings.right_relay, state);
-  }
-
-  return STATUS_CODE_OK;
+  return sequenced_relay_set_state(storage, state);
 }
 
-StatusCode motor_relay_init(MotorRelayStorage *storage, const MotorRelaySettings *settings) {
+StatusCode sequenced_relay_init(SequencedRelayStorage *storage,
+                                const SequencedRelaySettings *settings) {
   memset(storage, 0, sizeof(*storage));
   storage->settings = *settings;
   storage->delay_timer = SOFT_TIMER_INVALID_TIMER;
```

Comment:
Should we be cancelling an already running timer here?

Reply:
Nah, there's no way to tell if there's a previously valid timer since they could be passing in uninitialized memory. For unit testing, calling `soft_timer_init` will cancel any running timers.